chain/chain/src/state_snapshot_actor.rs

```diff
@@ -102,19 +124,23 @@ pub type MakeSnapshotCallback =
 pub fn get_make_snapshot_callback(
     state_snapshot_addr: Arc<actix::Addr<StateSnapshotActor>>,
     flat_storage_manager: FlatStorageManager,
-    compaction_enabled: bool,
 ) -> MakeSnapshotCallback {
     Arc::new(move |prev_block_hash, shard_uids, block| {
         tracing::info!(
             target: "state_snapshot",
             ?prev_block_hash,
             ?shard_uids,
             "start_snapshot_callback sends `MakeSnapshotCallback` to state_snapshot_addr");
-        if flat_storage_manager.set_flat_state_updates_mode(false) {
-            state_snapshot_addr.do_send(
-                MakeSnapshotRequest { prev_block_hash, shard_uids, block, compaction_enabled }
-                    .with_span_context(),
-            );
+        // We need to stop flat head updates synchronously in the client thread.
```

Comment:
Added some much needed documentation for why we are calling `set_flat_state_updates_mode` here